tools/validators/instance_validator/tests/entity_instance_test.py

```diff
@@ -288,5 +290,37 @@ def testValidateLinks(self):
     if not instance.IsValidEntityInstance():
       self.fail('exception incorrectly raised')
 
+  def testGoodConnections(self):
+    parsed = instance_parser.parse_yaml(
+        os.path.join(_TESTCASE_PATH,
+                     'GOOD',
+                     'good_building_connections.yaml'))
+    parsed = dict(parsed)
+    entity_name = list(parsed.keys())[0]
+
+    entity = dict(parsed[entity_name])
+    instance = entity_instance.EntityInstance(entity,
+                                              self.universe,
+                                              parsed.keys())
+
+    if not instance.IsValidEntityInstance():
+      self.fail('exception incorrectly raised')
+
```

Comment:
TODO:  Add tests for the more advanced syntax

Reply:
The syntax tests for connections already exist in a separate folder, and you can see them here: https://github.com/google/digitalbuildings/blob/master/tools/validators/instance_validator/tests/instance_parser_test.py.

Reply:
sorry, by syntax I meant the more advanced configuration.